pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -2272,6 +2282,8 @@ type SSSDSidecarAdditionalFile struct {
 }
 
 // NetworkSpec for Ceph includes backward compatibility code
+// +kubebuilder:validation:XValidation:rule="!has(self.provider) || (self.provider != 'multus' || (self.provider == 'multus' && size(self.selectors) > 0))",message="at least one network selector must be specified when using the multus"
+// +kubebuilder:validation:XValidation:rule="!has(self.provider) || (oldSelf.provider != 'host' || (oldSelf.provider == 'host' && self.provider != 'multus'))",message="invalid update: network provider from 'host' to 'multus' provider change is not allowed"
```

Comment:
This isn't a limitation that I have specified. But I think maybe this is based on Travis's limitation that is codified here right now:
https://github.com/rook/rook/blob/3c43268d0a924c3966a253b1bf28a5d491cdedb1/pkg/apis/ceph.rook.io/v1/network.go#L78-L87

I think the equivalent rule is to allow provider to transition from `""` to `"host"` and from `"host"` to `""`, but not other transitions.

@travisn is that right?

Along that trend, I wonder if it makes sense broadly disallow network transitions from provider to provider. This would enforce that all transitions go from `"providerA"` to `""` to `"providerB"`. Thoughts?

E.g.
```
// +kubebuilder:validation:XValidation:message="network provider must be disabled (reverted to empty string) before a new provider is enabled",rule="(oldSelf.provider == '' && self.provider != '') || (oldSelf.provider != '' && self.provider == '')"
```

Reply:
Yes, transitioning to and from host should be allowed. Transitioning between other providers is not supported, or at least questionable. Sounds good if we have a simple rule that says not to transition between providers.

Reply:
added the change and tested

Reply:
I didn't get the error when I can the `kubectl create -f cluster-multus-test.yaml` but in the operator logs I'm seeing the error.

Reply:
well came with more simpler rule to understand `rule="self == '' || self == oldSelf"` and tested it as well with watching operator logs, worked